barretenberg/cpp/src/barretenberg/crypto/merkle_tree/lmdb_store/callbacks.cpp

```diff
@@ -0,0 +1,148 @@
+#include "barretenberg/crypto/merkle_tree/lmdb_store/callbacks.hpp"
+#include "barretenberg/numeric/uint256/uint256.hpp"
+#include "lmdb.h"
+#include <algorithm>
+#include <cstdint>
+#include <cstring>
+#include <functional>
+#include <vector>
+
+namespace bb::crypto::merkle_tree {
+void throw_error(const std::string& errorString, int error)
+{
+    std::stringstream ss;
+    ss << errorString << ": " << error << " - " << mdb_strerror(error) << std::endl;
+    throw std::runtime_error(ss.str());
+}
+
+std::vector<uint8_t> serialise_key(uint8_t key)
+{
+    return { key };
+}
+
+void deserialise_key(void* data, uint8_t& key)
+{
+    uint8_t* p = static_cast<uint8_t*>(data);
+    key = *p;
+}
+
+std::vector<uint8_t> serialise_key(uint64_t key)
+{
+    const uint8_t* p = reinterpret_cast<uint8_t*>(&key);
+    return std::vector<uint8_t>(p, p + sizeof(key));
+}
+
+void deserialise_key(void* data, uint64_t& key)
+{
+    std::memcpy(&key, data, sizeof(key));
+}
+
+std::vector<uint8_t> serialise_key(uint128_t key)
+{
+    std::vector<uint8_t> buf(16);
+#ifdef __i386__
```

Comment:
A pity that you had to do this! Leaks implementation from uint128_t I guess?

Consider extending those classes to always have `data()` or something. Alternatively, I hope both paths are tested or it will come back to bite you ;)